### ISSUE TITLE
`credo:disable-for-lines:X` Not Working as Expected

### ISSUE THREAD

**Issue by kyleVsteger** [rrrene/credo#902]
### Environment

* Credo version (`mix credo -v`): `1.5.6-ref.main.771783b`
* Erlang/Elixir version (`elixir -v`): 
```
Erlang/OTP 24 [erts-12.0.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]
Elixir 1.12.1 (compiled with Erlang/OTP 22)
```
* Operating system: `MacOS 11.5.1`

### What were you trying to do?
Ignore this warning: `Function is too complex (CC is 11, max is 9).` by adding a `credo:disable-for-next-lines:8 Credo.Check.Refactor.CyclomaticComplexity`

The function is defined as:
```elixir
  @spec do_get_num_bucket(number()) :: String.t()
  defp do_get_num_bucket(num) do
    # credo:disable-for-lines:8 Credo.Check.Refactor.CyclomaticComplexity
    cond do
      num >= 0.2 and num <= 0.3 -> "pt2_pt3g"
      num >= 0.3 and num <= 0.4 -> "pt3_pt4g"
      num >= 0.4 and num <= 0.5 -> "pt4_pt5g"
      num >= 0.5 and num <= 1.0 -> "pt5_1g"
      num >= 1 -> "over_1g"
      true -> "na_gs"
    end
  end
```

### Expected outcome
The warning is suppressed

### Actual outcome
The warning persists

### Additional
I've used credo ignore in a similar context before with this line:
`# credo:disable-for-lines:7 Credo.Check.Refactor.Nesting`

When I add `# credo:disable-for-this-file Credo.Check.Refactor.CyclomaticComplexity` the warning is suppressed

**Comment by ckoch-cars** [comment]
I can confirm a similar behavior, if I move the inline comment `# credo:disable-for-lines:747 Credo.Check.Readability.MaxLineLength` around near the function (above and below the spec) the comment directive is ignored and the MaxLineLength check is made and  the check fails.  If I move the inline comment to line 2 below defmodule, then the comment is obeyed and the check is ignored. 

It seems plausible this may be related to https://github.com/rrrene/credo/issues/918
The file I am testing also has sigils (`~E` block) present and the behavior of the comment seems dependent on being read before some block. The file also has a module-level text block assigned to an anonymous variable. 
Ex (documentation for a private fn) 
```
defmodule D.E.F
  def my_fns
  _docp = """
    ...
  """
  defp my_priv ...
end
```

If I move the `# credo:disable-for-lines:xxx` below this `_docp =` block, then it is ignored. 

Environment
Credo version (mix credo -v): 1.6.1 sha: 7dc76dcd
Erlang/Elixir version (elixir -v):
Erlang/OTP 24 [erts-12.0.3] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit]
Elixir 1.13.1 (compiled with Erlang/OTP 24) 
Erlang 24.0.5
Operating system: MacOS 11.5.2

**Comment by rrrene** [comment]
Thx for reporting this! :+1: It should be be fixed in the latest version of Credo.

If it is not, please feel free to re-open this issue!
